src/sentry/services/hybrid_cloud/tombstone/impl.py

```diff
@@ -23,17 +25,32 @@ class MonolithTombstoneService(TombstoneService):
     # In the future, no single deployment can be a source of truth about the location of all models due to
     # deployment drift, however the current deployment (for which this is a bridging implementation), can.
     # We use silo limits information to infer the correct destination.
+    by_table_name: MutableMapping[str, Any]
 
-    def record_remote_tombstone(self, tombstone: ApiTombstone) -> None:
+    def __init__(self) -> None:
+        super().__init__()
+        self.by_table_name = {}
+
+    def _get_model(self, table_name: str) -> Any:
         from django.apps import apps
 
+        if model := self.by_table_name.get(table_name):
+            return model
+
         for app, app_models in apps.all_models.items():
             for model in app_models.values():
-                if limits := getattr(getattr(model, "_meta"), "silo_limits", None):
-                    if SiloMode.CONTROL in limits.silos:
-                        RegionTombstone.record_delete(tombstone.table_name, tombstone.identifier)
-                    else:
-                        ControlTombstone.record_delete(tombstone.table_name, tombstone.identifier)
+                if model._meta.db_table == table_name:
```

Comment:
If we also had the `app_label` on the tombstone we would be able to use `apps.get_model()`. Should we add the label as well?

Reply:
It is highly likely that the sentry application will continue to evolve in the future.  To the point that any historical `app_name` isn't necessarily going to be stable, imo.  You can say the same for `table_name` but if we have to have either one, or two, things be stable, I'd argue the former not the latter.  Plus, adding additional string column here suggests the dreadful case of actually having duplicate table_names across two different application name spaces.

In one iteration, I even considered only using integer enums and not strings, but considered that a step too far.

Ultimately, I didn't want to tie this aspect of the system too strongly to current django semantics.